internal/rbd/nodeserver.go

```diff
@@ -718,8 +719,20 @@ func (ns *NodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 	}
 
 	diskMounter := &mount.SafeFormatAndMount{Interface: ns.mounter, Exec: utilexec.New()}
-	// TODO check size and return success or error
 	volumePath += "/" + volumeID
+
+	// check size and return success or error
+	var vStat syscall.Statfs_t
+	err = syscall.Statfs(volumePath, &vStat)
+	if err != nil {
+		return nil, status.Errorf(codes.Aborted, util.VolumePathStatfsFailedFmt, req.GetVolumePath())
+	}
+	volSize := util.RoundOffBytes(req.GetCapacityRange().GetRequiredBytes())
+	vPathSize := util.RoundOffBytes(vStat.Bsize * int64(vStat.Blocks))
+	if vPathSize == volSize {
```

Comment:
we should return success if its already resized to new size

Reply:
replied in Humble's comment as both are same.